### PR TITLE
Convert BitHdtv to AngleSharp

### PR DESCRIPTION
**Pull request by cadatoiva**
Untested. #7292

### CODE REVIEW

File: src/Jackett.Common/Indexers/BitHdtv.cs [hunk 1]
```diff
@@ -134,48 +139,52 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
+                {
+                    var rows = table.Children;
                     foreach (var row in rows.Skip(1))
                     {
                         var release = new ReleaseInfo();
 
-                        var qRow = row.Cq();
-                        var qLink = qRow.Children().ElementAt(2).Cq().Children("a").First();
+                        
+                        var qLink = row.Children.ElementAt(2).QuerySelectorAll("a").First();
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/BitHdtv.cs]: var qLink = row.Children[2].QuerySelector("a");

File: src/Jackett.Common/Indexers/BitHdtv.cs [hunk 2]
```diff
@@ -134,48 +139,52 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
-                        release.Guid = new Uri(qLink.Attr("href"));
+                        release.Files = ParseUtil.CoerceLong(row.QuerySelector("td:nth-child(4)").Text());
+                        release.Grabs = ParseUtil.CoerceLong(row.QuerySelector("td:nth-child(8)").Text());
+                        release.Guid = new Uri(qLink.GetAttribute("href"));
                         release.Comments = release.Guid;
-                        release.Link = new Uri(string.Format(DownloadUrl, qLink.Attr("href").Split('=')[1]));
+                        release.Link = new Uri(string.Format(DownloadUrl, qLink.GetAttribute("href").Split('=')[1]));
 
-                        var catUrl = qRow.Children().ElementAt(1).FirstElementChild.Cq().Attr("href");
+                        var catUrl = row.Children.ElementAt(1).FirstElementChild.GetAttribute("href");
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/BitHdtv.cs]: row.Children[1]
There are other cases below

File: src/Jackett.Common/Indexers/BitHdtv.cs [hunk 3]
```diff
@@ -134,48 +139,52 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
-                CQ dom = results.Content;
-                dom["#needseed"].Remove();
-                foreach (var table in dom["table[align=center] + br + table > tbody"])
+                var parser = new HtmlParser();
+                var dom = parser.ParseDocument(results.Content);
+                foreach (var child in dom.QuerySelectorAll("#needseed"))
                 {
-                    var rows = table.Cq().Children();
+                    dom.RemoveChild(child);
+                }
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/BitHdtv.cs]: remove braces

File: src/Jackett.Common/Indexers/BitHdtv.cs [hunk 4]
```diff
@@ -134,48 +139,52 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
+                {
+                    var rows = table.Children;
                     foreach (var row in rows.Skip(1))
                     {
                         var release = new ReleaseInfo();
 
-                        var qRow = row.Cq();
-                        var qLink = qRow.Children().ElementAt(2).Cq().Children("a").First();
+                        
+                        var qLink = row.Children[2].QuerySelectorAll("a").First();
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/BitHdtv.cs]: var qLink = row.Children[2].QuerySelector("a");

File: src/Jackett.Common/Indexers/BitHdtv.cs [hunk 5]
```diff
@@ -110,9 +110,9 @@ public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken
                 var dom = parser.ParseDocument(response.Content);
                 var messageEl = dom.QuerySelectorAll("table.detail td.text").Last();
                 foreach (var child in messageEl.QuerySelectorAll("a"))
-                    messageEl.RemoveChild(child);
+                    child.Remove();
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/BitHdtv.cs]: @cadatoiva I hope we are not breaking anything removing elements while iterating. This is not safe in other languages.

Comment (cadatoiva) [reply src/Jackett.Common/Indexers/BitHdtv.cs]: This is the recommended way to remove elements according to the AngleSharp website:

https://anglesharp.github.io/docs/Questions.html#how-can-i-remove-all-elements-matching-a-certain-selector

**Pull request merged**